### PR TITLE
[4.0] Updating all views to use H1 for Component Title when Menu Title is hidden

### PR DESCRIPTION
**Pull request by AndySDH**
Follow-up Pull Request after #32448

### Summary of Changes

After updating the article view with the above-mentioned PR, this PR updates all the other views as well to show the component title as H1 when the Menu Title is hidden.

In particular this updates the following views:

- Contact Default
- Content Category Default
- Content Category Blog
- Newsfeed Category
- Tag Default
- Tag List 

In addition, when doing this, I stumbled upon an unused page_subheading param that was present in the 'featured' views and the 'category blog' views. This looks to be an outdated and out of place parameter that was present just in those two views and nowhere else. I don't think nobody has even ever used it.

So this also addresses issue #29586, removing the page_subheading parameter in those views, making them consistent with all the other views as well.

### Actual result BEFORE applying this Pull Request

Components title is H2.

### Expected result AFTER applying this Pull Request

Components title is H1 when Menu Title is hidden.

### CODE REVIEW

**Comment by richard67** [comment]
@AndySDH So this PR solves that issue #29586 ?

**Comment by AndySDH** [comment]
Please test @laoneo @bembelimen @brianteeman @simbus82 @Quy
Thanks!

File: components/com_tags/src/View/Tag/HtmlView.php [hunk 1]
```diff
@@ -313,10 +313,6 @@ protected function _prepareDocument()
 			$this->params->def('page_heading', $this->params->get('page_title', $menu->title));
 			$title = $title ?: $this->params->get('page_title', $menu->title);
```
Comment (richard67) [inline components/com_tags/src/View/Tag/HtmlView.php]:
```suggestion
```
Please remove this line, too, so the PHP code style linter is happy: https://ci.joomla.org/joomla/joomla-cms/40397/1/6

**Comment by AndySDH** [comment]
> @AndySDH So this PR solves that issue #29586 ?

Yeah, addresses that too as explained above. I opted to just remove the parameter as it was widely inconsistent and out of place (especially in the blog view). I don't think nobody has even ever used it.

**Comment by richard67** [comment]
> Yeah, addresses that too as explained above :)

@AndySDH Yes, but to address and to solve it is not 100 % the same thing, so I asked to be sure. Thanks for reporting back, and thanks for your contributions so far, it's really appreciated.

File: components/com_content/tmpl/featured/default.xml [hunk 2]
```diff
@@ -174,14 +174,6 @@
 				<option value="1">JSHOW</option>
 			</field>
 
-			<field
-				name="page_subheading"
-				type="text"
-				label="JGLOBAL_SUBHEADING_LABEL"
```
Comment (Quy) [inline components/com_content/tmpl/featured/default.xml:180]: Delete language strings.

Comment (AndySDH) [reply components/com_content/tmpl/featured/default.xml:180]: Done, thanks!

**Comment by himanshu007-creator** [comment]
I have tested this item :white_check_mark: successfully on 970c03dd9ad8416daa13eb57a1f455533236700b<br /><br />I opened the Contact default view. Before applying the patch title was having **h2** tag, after patch, it was **h1**.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

File: components/com_content/tmpl/category/blog.php [hunk 3]
```diff
@@ -37,12 +39,10 @@
 		</div>
 	<?php endif; ?>
 
-	<?php if ($this->params->get('show_category_title', 1) or $this->params->get('page_subheading')) : ?>
-		<h2> <?php echo $this->escape($this->params->get('page_subheading')); ?>
-			<?php if ($this->params->get('show_category_title')) : ?>
-				<span class="subheading-category"><?php echo $this->category->title; ?></span>
-			<?php endif; ?>
-		</h2>
+	<?php if ($this->params->get('show_category_title', 1): ?>
```
Comment (particthistle) [inline components/com_content/tmpl/category/blog.php]: PR breaks for Category Blog page due to missing )

Line should be:

`<?php if ($this->params->get('show_category_title', 1)) : ?>`

**Comment by particthistle** [comment]
I have tested this item :red_circle: unsuccessfully on 970c03dd9ad8416daa13eb57a1f455533236700b<br /><br />See next comment and review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Review by particthistle — CHANGES_REQUESTED** [review]
@AndySDH - working through testing all the scenarios (@himanshu007-creator - will chat in Glip about improving the level of detail for your testing) and found a syntax error that breaks the Blog Category Layout when the patch is applied.

Aside from that, all the other scenarios look to have worked correctly, so will be able to retest quickly when this has been fixed up as I've got all the menu items for the different scenarios saved and ready to go.

**Comment by infograf768** [comment]
This one also needs changes:

https://github.com/joomla/joomla-cms/blob/90ab8c02dbb9a0763e9b8bb91bfea690cbc1dd44/plugins/sampledata/multilang/multilang.php#L856-L869

would also be modified to include the changes as done in https://github.com/joomla/joomla-cms/pull/32550

**Comment by AndySDH** [comment]
> @AndySDH - working through testing all the scenarios (@himanshu007-creator - will chat in Glip about improving the level of detail for your testing) and found a syntax error that breaks the Blog Category Layout when the patch is applied.
> 
> Aside from that, all the other scenarios look to have worked correctly, so will be able to retest quickly when this has been fixed up as I've got all the menu items for the different scenarios saved and ready to go.

Whoops, sorry, thanks for reporting this! It's now fixed.

How did the auto-check not automatically spot this?

File: components/com_content/tmpl/category/blog.php (hunk 3 above)
Comment (AndySDH) [reply components/com_content/tmpl/category/blog.php]: Fixed, thanks!

**Comment by particthistle** [comment]
@AndySDH, take a look at @infograf768 suggest change needed for the sample data.

#32550 is being worked on concurrently, but it's taking care of the default install. The Sample Data has the same problem that PR is looking to fix, which is that the sample data has J3 parameters not J4 parameters. Just going to see if I can work out what needs to go in there quickly to speed up the change for you.

Once that's also fixed I'll then retest also with a new install + sample data install.

File: plugins/sampledata/multilang/multilang.php [hunk 4]
```diff
@@ -856,7 +856,7 @@ private function addBlogMenuItem($itemLanguage, $categoryId)
 			'params'       => '{"layout_type":"blog","show_category_heading_title_text":"","show_category_title":"",'
 				. '"show_description":"","show_description_image":"","maxLevel":"","show_empty_categories":"",'
 				. '"show_no_articles":"","show_subcat_desc":"","show_cat_num_articles":"","show_cat_tags":"",'
```
Comment (particthistle) [inline plugins/sampledata/multilang/multilang.php:858]: To correct this data in regards to #32550 another parameter needs to be added before num_leading_articles:
Add before line 859:
`. '"blog_class_leading":"","blog_class":"",'`

File: plugins/sampledata/multilang/multilang.php [hunk 5]
```diff
@@ -856,7 +856,7 @@ private function addBlogMenuItem($itemLanguage, $categoryId)
...
-				. '"page_subheading":"","num_leading_articles":"1","num_intro_articles":"3",'
+				. '"num_leading_articles":"1","num_intro_articles":"3",'
 				. '"num_links":"0","show_subcategory_content":"","orderby_pri":"",'
```
Comment (particthistle) [inline plugins/sampledata/multilang/multilang.php]: Addition for #32550
Line 860 - extra parameter link_intro_image
`. '"num_links":"","show_subcategory_content":"","link_intro_image":"","orderby_pri":"",'`

File: plugins/sampledata/multilang/multilang.php [hunk 6]
```diff
@@ -856,7 +856,7 @@ private function addBlogMenuItem($itemLanguage, $categoryId)
...
 				. '"orderby_sec":"front","order_date":"","show_pagination":"2","show_pagination_results":"1",'
 				. '"show_featured":"","show_title":"","link_titles":"","show_intro":"","info_block_position":"",'
```
Comment (particthistle) [inline plugins/sampledata/multilang/multilang.php]: Addition for #32550
Line 862 - extra parameter article_layout
`. '"show_featured":"","article_layout":"_:default","show_title":"","link_titles":"","show_intro":"","info_block_position":"",'`

**Review by particthistle — CHANGES_REQUESTED** [review]
Changes to match up #32550. Part 1.

**Comment by particthistle** [comment]
Changes to match up #32550. Part 2.

It stopped letting me add reviews to the code after line 862 for some reason. There's one more line needing adjustment to match up the changes outlined in #32550.

~Line 864 - remove show_associations parameter: New line:
`. '"link_parent_category":"","show_author":"","link_author":"",'`~

@andySDH Ignore this per @infograf768 comment below.

**Comment by infograf768** [comment]
`show_associations` should stay as it is a blog parameter (Options).

Screen Shot 2021-03-02 at 11 10 03

**Comment by AndySDH** [comment]
Thanks for the feedback, I'll wait for #32550 to be finalized and merged, and once that's merged, I'll make sure to have the same version of those files in this PR as well.

**Comment by ceford** [comment]
#32550 is marked as merged - so is this ready for testing?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Comment by richard67** [comment]
@ceford I think maybe @AndySDH wants to update his branch first to latest 4.0-dev so it contains the changes from the merged PR, too. Let's see what he replies.

**Comment by AndySDH** [comment]
Resolved conflicts with #32550 :)

**Comment by particthistle** [comment]
I have tested this item :red_circle: unsuccessfully on 0f22848e688e6f2e5e8c8c6ce12d552d45dbcd43<br /><br />1. Syntax error with Blog Category Layout fixed from previous test, so this part was successful.

2. Additional changes were suggested for plugins/sampledata/multilang/multilang.php but have not been applied, so the parameters don't match #32550.

https://github.com/joomla/joomla-cms/pull/32525#discussion_r585126993 - Add `. '"blog_class_leading":"","blog_class":"",'` before line 859.
https://github.com/joomla/joomla-cms/pull/32525#discussion_r585128962 - Add `"link_intro_image":"",` at 860
https://github.com/joomla/joomla-cms/pull/32525#discussion_r585129380 - Add `"article_layout":"_:default",` at 862


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Comment by AndySDH** [comment]
@particthistle Done, thanks!

**Comment by particthistle** [comment]
I have tested this item :white_check_mark: successfully on e94a87a0611b4f0023d28207c799616240cd6a37<br /><br />Thanks @AndySDH 

- H1 settings all working.

- Changes to related items identified in the multi-language sample data identified in #32550 also now applying successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Comment by chmst** [comment]
Concerning subheading: It was used in 3.9 https://issues.joomla.org/tracker/joomla-cms/30422<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Comment by AndySDH** [comment]
> Concerning subheading: It was used in 3.9 https://issues.joomla.org/tracker/joomla-cms/30422
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/32525](https://issues.joomla.org/tracker/joomla-cms/32525).

I'm aware but it was used present only in 1 view (2 if you count Featured view), and absent in all the others. It was wildly inconsistent and I felt it served not much of a purpose. So I thought it made it sense to clean it up and remove it.

**Comment by particthistle** [comment]
Subheading removal: Should it go ahead (I don't know anyone using it, but I'm sure there may be) do we just need to make sure it's listed here: https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4

If so, add Documentation Required tag for follow up.

**Comment by himanshu007-creator** [comment]
I have tested this item :white_check_mark: successfully on e94a87a0611b4f0023d28207c799616240cd6a37<br /><br />Tested the behavior on all mentioned categories. Thanks @particthistle for helping out earlier.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Comment by richard67** [comment]
I've allowed myself to solve the merged conflict and restored the previous test results because the merge after that for resolving the merge conflict was only a code style change.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32525">issues.joomla.org/tracker/joomla-cms/32525</a>.</sub>

**Comment by AndySDH** [comment]
Thanks @richard67

**Comment by chmst** [comment]
Thanks!

**Pull request merged**